csharp/src/Client/AdbcDataReader.cs

```diff
@@ -0,0 +1,324 @@
+        {
+            return this.recordBatch.Schema.GetFieldIndex(name);
+        }
+
+        public override string GetString(int ordinal)
+        {
+            return Convert.ToString(GetValue(ordinal));
+        }
+
+        public override object GetValue(int ordinal)
+        {
+            return GetValue(this.recordBatch?.Column(ordinal), ordinal);
+        }
+
+        public override int GetValues(object[] values)
+        {
+            int count = Math.Min(FieldCount, values.Length);
+            for (int i = 0; i < count; i++)
+            {
+                values[i] = GetValue(i);
+            }
+            return count;
+        }
+
+        public override bool IsDBNull(int ordinal)
+        {
+            return GetValue(ordinal) == null;
+        }
+
+        public override bool NextResult()
+        {
+            this.recordBatch = ReadNextRecordBatchAsync().Result;
+
+            if (this.recordBatch != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public new void Dispose()
+        {
+            this.recordBatch?.Dispose();
+        }
+
+        public override bool Read()
+        {
+            if (this.recordBatch != null && this.currentRowInRecordBatch < this.recordBatch.Length - 1)
+            {
+                this.currentRowInRecordBatch++;
+                return true;
+            }
+
+            this.recordBatch = ReadNextRecordBatchAsync().Result;
+
+            return this.recordBatch != null;
+        }
+
+        public override DataTable GetSchemaTable()
+        {
+            if (this.schema != null)
+            {
+                return SchemaConverter.ConvertArrowSchema(this.schema, this.adbcCommand.AdbcStatement);
+            }
+            else
+            {
+                throw new InvalidOperationException("Cannot obtain schema table because the Arrow schema is null");
+            }
+        }
+
+#if NET5_0_OR_GREATER
+        public override Task<DataTable> GetSchemaTableAsync(CancellationToken cancellationToken = default)
+        {
+            return Task.Run(() => GetSchemaTable(), cancellationToken);
+        }
+#endif
+
+        public ReadOnlyCollection<DbColumn> GetColumnSchema()
+        {
+            if (this.schema != null)
+            {
+                List<DbColumn> dbColumns = new List<DbColumn>();
+
+                foreach (Field f in this.schema.FieldsList)
+                {
+                    Type t = SchemaConverter.ConvertArrowType(f);
+
+                    if(
+                        f.DataType.TypeId == Types.ArrowTypeId.Double ||
+                        f.DataType.TypeId == Types.ArrowTypeId.Float ||
+                        f.DataType.TypeId == Types.ArrowTypeId.Decimal128 ||
+                        f.DataType.TypeId == Types.ArrowTypeId.Decimal256
+                    )
+                    {
+                        int precision = Convert.ToInt32(f.Metadata["precision"]);

```

Comment:
For this part. Snowflake NUMBER type also has precision and scale. And f.DataType.TypeId == Types.ArrowTypeId.Int64.  User can define NUMBER(38,0), NUMBER(25,0) and In this case, the precision and scale are lost. 

I don't know what's the behavior for other database. How about we just keep 'precision' and 'scale' if the precision and scale is existed? In this way, in this way we don't lose any metadata.

Reply:
Thanks for the feedback. I made this change and welcome additional contributions.